editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -787,7 +790,11 @@ impl<'a> MessageHandler<NodeGraphMessage, NodeGraphHandlerData<'a>> for NodeGrap
 			NodeGraphMessage::SetNodeInput { node_id, input_index, input } => {
 				if let Some(network) = document.document_network.nested_network_mut(&self.network) {
 					if let Some(node) = network.nodes.get_mut(&node_id) {
-						node.inputs[input_index] = input
+						let structure_changed = node.inputs[input_index].as_node().is_some() || input.as_node().is_some();
```

Comment:
Is it theoretically possible this indexing could go out of bounds and panic? Do you think `.get()` would be a worthwhile defensive programming measure?